tensorflow_datasets/scripts/cleanup/delete_old_versions.py

```diff
@@ -0,0 +1,111 @@
+# coding=utf-8
+# Copyright 2020 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""Script to detect all the versions of the various datasets and delete the old/non-existing ones
+
+User is first given previews of all the versions of the datasets with the latest ones in bold
+
+Instructions:
+
+python -m tensorflow_datasets.scripts.delete_old_versions \
+  --data_dir=/path/to/data_dir/
+
+"""
+
+import os
+from absl import flags
+from absl import app
+
+import tensorflow_datasets as tfds
+from tensorflow_datasets.core import constants
+from tensorflow_datasets.core.load import list_full_names
+
+import tensorflow as tf
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string("data_dir",constants.DATA_DIR,"Path to the data directory")
+
+def get_redundant_datasets(data_dir):
+  f"""
+  Returns a tuple of installed datasets and rogue datasets
+
+  Arguments:
+      data_dir : The path to the data directory
+
+  Returns:
+      installed_datasets, rogue_datasets : Tuple of installed and non-existing datasets
+  """
+  installed_datasets = []
+  all_datasets = tfds.list_builders()
+  exclude = {"downloads","download","manual","extracted"}
+
+  #Get all the non-existing datasets
+  rogue_datasets = []
+  for dataset in tf.io.gfile.listdir(data_dir):
+    if dataset not in all_datasets:
+      rogue_datasets.append(dataset)
+  rogue_datasets.remove("downloads")
+
+  #Finding all installed datasets
+  for root,dirs,_ in tf.io.gfile.walk(data_dir, topdown=True):
+    #Excluding the downloads directory and the rogue datasets
+    dirs[:] = [d for d in dirs if d not in exclude]
+    dirs[:] = [d for d in dirs if d not in rogue_datasets]
+    if dirs==[]:
+      installed_datasets.append(os.path.relpath(root,data_dir).replace("\\","/"))
```

Comment:
`os.path.normpath`, also note there is `os.path.expanduser` to expand remove the `~/`

Reply:
@Conchylicultor `os.path.normpath` replaces all the path separators with `'\\'` on windows system whereas the `tfds.core.load.list_full_names` returns list of datasets with `'/'` as the separator due to which the dataset names cannopt be compared. Could you give me some pointers? Thanks!

Reply:
Then you could use `os.path.normpath` on `tfds.core.load.list_full_names` to convert path into windows supported paths